packages/react-components/src/components/Select/Select.tsx

```diff
@@ -0,0 +1,371 @@
+  error?: string;
+  id: string;
+  controlledIsOpen?: boolean;
+  getItemBody: (props: ISelectItem['props']) => React.ReactNode;
+  getSelectedItemBody: (props: ISelectItem['props']) => React.ReactNode;
+  onItemSelect: (itemKey: string) => void;
+  searchEmptyState?: React.ReactNode;
+  items: ISelectItem[];
+  searchPlaceholder?: string;
+  searchProperty: string | string[];
+  selected: string | number | null;
+  search?: boolean;
+  required?: boolean;
+  placeholder?: string;
+  disabled?: boolean;
+  openedOnInit?: boolean;
+  onDropdownToggle?: (v: boolean) => void;
+  onSearchPhraseChange?: (phrase: string) => void;
+  selectHeader?: string;
+}
+
+export const Select: React.FC<ISelectProps> = ({
+  className,
+  error,
+  id,
+  controlledIsOpen,
+  getItemBody,
+  getSelectedItemBody,
+  onItemSelect,
+  searchEmptyState,
+  items = [],
+  searchPlaceholder,
+  searchProperty,
+  selected,
+  search,
+  required,
+  placeholder,
+  disabled,
+  openedOnInit,
+  onDropdownToggle = () => null,
+  onSearchPhraseChange,
+  selectHeader,
+}) => {
+  let timerId: ReturnType<typeof setTimeout> | null = null;
+  const [isFocused, setIsFocused] = React.useState(false);
+  const [searchPhrase, setSearchPhrase] = React.useState('');
+  const [isOpen, setIsOpen] = React.useState(
+    openedOnInit || controlledIsOpen || false
+  );
+  const [focusedItemKey, setFocusedItemKey] = React.useState(
+    items[0] ? items[0].key : null
+  );
+  const containerRef = React.createRef<HTMLDivElement>();
+  const searchInputRef = React.createRef<HTMLInputElement>();
+  const headRef = React.createRef<HTMLDivElement>();
+  const clearButtonRef = React.createRef<HTMLDivElement>();
+  const listRef = React.createRef<HTMLUListElement>();
+
+  const isIsOpenControlled = () => controlledIsOpen !== undefined;
+
+  const getIsOpen = () => {
+    if (disabled) {
+      return false;
+    }
+
+    return isIsOpenControlled() ? controlledIsOpen : isOpen;
+  };
+
+  const onDocumentClick = (e: MouseEvent) => {
+    if (
+      getIsOpen() &&
+      containerRef.current &&
+      !containerRef.current.contains(e.target as Node) &&
+      listRef?.current
+    ) {
+      listRef.current.scrollTop = 0;
+      hideSelectBody();
+    }
+  };
+
+  const onBodyOpen = () => {
+    document.addEventListener('click', onDocumentClick);
+
+    if (search && searchInputRef.current) {
+      timerId = setTimeout(() => {
+        searchInputRef?.current?.focus();
+      }, 150);
+    }
+  };
+
+  const onBodyClose = () => {
+    document.removeEventListener('click', onDocumentClick);
+
+    if (timerId) {
+      clearTimeout(timerId);
+    }
+  };
+
+  React.useEffect(() => {
+    if (isOpen) {
+      onDropdownToggle(true);
+      onBodyOpen();
+    }
+  }, []);
+
+  React.useEffect(() => {
+    if (getIsOpen()) {
+      return onBodyOpen();
+    }
+
+    if (!getIsOpen()) {
+      onBodyClose();
+    }
+  }, [getIsOpen]);
+
+  const filterItem = (item: any): any => {
```

Comment:
Are you sure that this function returns `any`? My guess is `boolean`. And it would be good to set typing for `item` also.